### PR TITLE
Edit existing slide notes and add notes to slides which don't already have them

### LINKED ISSUE #684: Add Slide Note
### Discussed in https://github.com/ShapeCrawler/ShapeCrawler/discussions/682

<div type='discussions-op-text'>

<sup>Originally posted by **jcoliz** May 14, 2024</sup>
Is it possible to add/edit slide notes? I could not find this, if exists.</div>

### LINKED ISSUE #685: Edit Slide Note
### Discussed in https://github.com/ShapeCrawler/ShapeCrawler/discussions/682

<div type='discussions-op-text'>

<sup>Originally posted by **jcoliz** May 14, 2024</sup>
Is it possible to add/edit slide notes? I could not find this, if exists.</div>

### PR DESCRIPTION
**Pull request by jcoliz**
Fixes #684 
Fixes #685

---

## PR-Codex overview
This PR enhances ShapeCrawler by adding support for slide notes. 

### Detailed summary
- Added `Notes` property to `ISlide` interface.
- Implemented `AddNotes` method in `Slide` class.
- Added unit tests for `Notes` getter, `AddNotes` method, and various scenarios.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by jcoliz** [comment]
Test failures look like a difference in meaning of `Environment.Newline` between Windows and Linux. Will improve this.

**Comment by jcoliz** [comment]
@ashahabov Suggest to disable [csharpsquid:S3220](https://sonarcloud.io/organizations/shapecrawler/rules?open=csharpsquid%3AS3220&rule_key=csharpsquid%3AS3220) in SonarCloud.

The OpenXML library is filled with cases where constructors take `params OpenXmlElement[]` OR `IEnumerable<OpenXmlElement>`. Given that an `OpenXmlElement` *implements* `IEnumerable<OpenXmlElement>`, ANY time we pass a single `OpenXmlElement` to any constructor, this warning will get fired.

Working around this to pass SonarCloud resulted in *far less* readable code, IMO.

File: src/ShapeCrawler/Presentations/ISlide.cs [hunk 1]
```diff
@@ -66,6 +71,11 @@ public interface ISlide
     /// </summary>
     ITable TableWithName(string table);
     
+    /// <summary>
+    ///     Adds a notes slide if there isn't already one.
+    /// </summary>
+    void AddNotesIfEmpty();
```
Comment (ashahabov) [inline src/ShapeCrawler/Presentations/ISlide.cs:77]: How about renaming and slightly changing the method signature to `AddNotes(string notes)`?

As mentioned earlier, the idea is to make the use of the high-level library's API closer to the PowerPoint user experience.

File: test/ShapeCrawler.Tests.Unit/TextFrameTests.cs [hunk 2]
```diff
@@ -505,5 +505,134 @@ public void BottomMargin_getter_returns_bottom_margin_of_text_frame_in_centimete
             // Assert
             bottomMargin.Should().Be((decimal)expectedMargin);
         }
+
+        [Test]
+        public void SlideNotes_getter_returns_notes()
```
Comment (ashahabov) [inline test/ShapeCrawler.Tests.Unit/TextFrameTests.cs:510]: I think the more suitable place for these tests is `SlideTests.cs` with naming `Notes_Getter_returns_notes()`.

File: test/ShapeCrawler.Tests.Unit/TextFrameTests.cs [hunk 3]
```diff
@@ -505,5 +505,134 @@ public void BottomMargin_getter_returns_bottom_margin_of_text_frame_in_centimete
...
+            var expected = "SlideAddNotes_can_change_notes";
+
+            // Act
+            notes.Text = expected;
+
+            // Assert
+            notes.Text.Should().Be(expected);
+            pres.Validate();
+
+            // Just in case you want to check it!
```
Comment (ashahabov) [inline test/ShapeCrawler.Tests.Unit/TextFrameTests.cs:608]: Please remove these comments.

I think we should document such guides in the out of the code comment, for example, in [CONTRIBUTING.md](https://github.com/ShapeCrawler/ShapeCrawler/blob/master/CONTRIBUTING.md).

You can also use the test method [SaveResult(pres)](https://github.com/ShapeCrawler/ShapeCrawler/blob/364bd96d407c43377105036b995d1d520b4a7451/test/ShapeCrawler.Tests.Unit.xUnit/Helpers/SCTest.cs#L63) which saves the test results in `\test\TestResults`

**Review by ashahabov — CHANGES_REQUESTED** [review]

**Comment by ashahabov** [comment]
> @ashahabov Suggest to disable [csharpsquid:S3220](https://sonarcloud.io/organizations/shapecrawler/rules?open=csharpsquid%3AS3220&rule_key=csharpsquid%3AS3220) in SonarCloud.
> 
> The OpenXML library is filled with cases where constructors take `params OpenXmlElement[]` OR `IEnumerable<OpenXmlElement>`. Given that an `OpenXmlElement` _implements_ `IEnumerable<OpenXmlElement>`, ANY time we pass a single `OpenXmlElement` to any constructor, this warning will get fired.
> 
> Working around this to pass SonarCloud resulted in _far less_ readable code, IMO.

Done. This rule has been deactivated.

File: test/ShapeCrawler.Tests.Unit/TextFrameTests.cs (hunk 2 above)
Comment (jcoliz) [reply test/ShapeCrawler.Tests.Unit/TextFrameTests.cs]: Done in latest

File: src/ShapeCrawler/Presentations/ISlide.cs (hunk 1 above)
Comment (jcoliz) [reply src/ShapeCrawler/Presentations/ISlide.cs]: Ok, I changed to take enumerable of strings, as would be better to add all the paragraphs at once if we're adding many.

**Pull request merged**